dependencies.yaml

```diff
@@ -194,12 +178,9 @@ dependencies:
       - output_types: [pyproject, requirements]
         matrices:
           - matrix: {cuda: "12.*"}
-            packages:
+            packages: &cupy_packages_cu12
               - cupy-cuda12x>=12.0.0
-          - matrix: {cuda: "11.*"}
-            packages: &cupy_packages_cu11
-              - cupy-cuda11x>=12.0.0
-          - {matrix: null, packages: *cupy_packages_cu11}
+          - {matrix: null, packages: *cupy_packages_cu12}
```

Comment:
What is this last matrix used for?

Wondering given we just have CUDA 12 above

Reply:
We have to have a “fallback” matrix for the default pyproject.toml output, which doesn’t specify a CUDA version. We typically default to unsuffixed RAPIDS libraries (like “cudf”) but suffixed CuPy libraries (like “cupy-cuda12x”).

Reply:
I'm too late now, but in the future, I would prefer to be explicit about the CUDA major version handling if possible. I think we do that more often than not. Or if you have a strong preference here, let's just make it a style guide rule (https://github.com/rapidsai/build-planning/blob/main/docs/docs/packaging.md#dependenciesyaml-conventions) and implement it across all repos.

Reply:
Sorry Bradley

Kyle added a comment with the CUDA version in all such sections like so

https://github.com/rapidsai/nx-cugraph/blob/a8c5bf8b651c2eb2c16aa8afd2ac1b45416b1f1d/dependencies.yaml#L178-L182

Does that help with this concern?